### PR TITLE
tls: cleaning up unused noboring code

### PR DESCRIPTION
**Pull request by alyssawilk**
https://github.com/envoyproxy/envoy/pull/32748 was an attempt to move no-boring code to contrib.
Per investigation at that time, the noboring build was already broken, so cleaning up that code entirely.
If needed, https://github.com/envoyproxy/envoy/pull/32748 can be reinstated instead.

This also cleans up some unneeded DNS code.

Risk Level: low
Testing:  n/a
Docs Changes: n/a
Release Notes: n/a
https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/extensions/all_extensions.bzl [hunk 1]
```diff
@@ -36,6 +36,9 @@ _core_extensions = [
     "envoy.network.dns_resolver.cares",
     "envoy.network.dns_resolver.apple",
     "envoy.load_balancing_policies.round_robin",
+    "envoy.http.original_ip_detection.xff",
+    "envoy.request_id.uuid",
```
Comment (ggreenway) [inline source/extensions/all_extensions.bzl:40]: These two seem unrelated. What's the story with them?

File: source/extensions/all_extensions.bzl [hunk 2]
```diff
@@ -36,6 +36,9 @@ _core_extensions = [
...
+    "envoy.transport_sockets.tls",
```
Comment (ggreenway) [inline source/extensions/all_extensions.bzl:41]: Given that all the real code is in `source/common/tls`, I'm not sure this is needed

File: source/server/BUILD [hunk 3]
```diff
@@ -465,6 +464,7 @@ envoy_cc_library(
         "//source/common/upstream:cluster_manager_lib",
         "//source/common/upstream:health_discovery_service_lib",
         "//source/common/version:version_lib",
+        "//source/extensions/transport_sockets/tls:config",
```
Comment (ggreenway) [inline source/server/BUILD:467]: Can this depend on `//source/common/tls:context_lib` instead?

**Review by ggreenway — CHANGES_REQUESTED** [review]
I like this

/wait

File: source/extensions/all_extensions.bzl (hunk 2 above)
Comment (alyssawilk) [reply source/extensions/all_extensions.bzl:41]: yeah I suspect with the reworking I can revert this.  I think it's a bug that we build code extensiosn but not required but arguably should go in a different PR

File: source/server/BUILD (hunk 3 above)
Comment (alyssawilk) [reply source/server/BUILD:467]: I think not without TLS being a required extension.  Have to do one or the other.  have a preference?

Comment (ggreenway) [reply source/server/BUILD:467]: I think it's cleaner to have the dependency her on source/common if possible. It's probably fine to make tls a required extension.

File: source/server/BUILD [hunk 4]
```diff
@@ -462,9 +461,10 @@ envoy_cc_library(
         "//source/common/signal:fatal_error_handler_lib",
         "//source/common/singleton:manager_impl_lib",
         "//source/common/stats:thread_local_store_lib",
+        "//source/common/tls:context_lib",
         "//source/common/upstream:cluster_manager_lib",
-        "//source/common/upstream:health_discovery_service_lib",
```
Comment (ggreenway) [inline source/server/BUILD:466]: I think this deletion was accidental, and you meant to remove tls:config two lines down, right?

**Review by ggreenway — CHANGES_REQUESTED** [review]

File: source/extensions/all_extensions.bzl (hunk 1 above)
Comment (ggreenway) [reply source/extensions/all_extensions.bzl:40]: Ping

**Review by ggreenway — APPROVED** [review]
